### ISSUE TITLE
Cannot configure build args using `tasks.json`

### ISSUE THREAD

**Issue by tudortimi** [microsoft/vscode-docker#3400]
I'm trying to add `--build-arg` to the `docker build` command triggered by the `Docker Images: Build Image...` task.

I created the following `.vscode/tasks.json` file:

```
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "docker-build",
            "dockerBuild": {
                "buildArgs": {
                    "SOME_ARG": "foo",
                },
            },
        }
    ],
}
```

I can reproduce the problem with a dummy `Dockerfile`:

```
FROM centos:7
```

When I execute the `Build Image` task, the following command gets called:

    docker build --pull --rm -f "Dockerfile" -t vscodedockerissue:latest "."

I expected the following to be called:

    docker build --build-arg SOME_ARG=foo --pull --rm -f "Dockerfile" -t vscodedockerissue:latest "."

VSCode: 1.63.2
Docker Extension: v1.19.0

**Comment by bwateratmsft** [comment]
This is due to the difference between the "Docker Images: Build Image..." command, and the `docker-build` task. Running that command does not actually run the task, unless a Dockerfile is explicitly set in the task, and that Dockerfile is chosen from the "Build Image" command. There's a couple different approaches you can take here:

1. The simplest approach is probably to just add the Dockerfile and some other required parameters to the task, basically this:

```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "docker-build",
            "dockerBuild": {
                "buildArgs": {
                    "SOME_ARG": "foo"
                },
                "tag": "vscodedockerissue",
                "context": "${workspaceFolder}",
                "dockerfile": "${workspaceFolder}/Dockerfile"
            }
        }
    ]
}
```

2. Another way to run the task is to do command "Tasks: Run Task", and then choose your `docker-build` task. You can also set it as the [default build task](https://code.visualstudio.com/docs/editor/tasks#_typescript-hello-world) in order to shortcut to <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd>.

The reason for this disparity is historical. The tasks are far newer than the command, and the tasks can only be used if they are defined and the Dockerfile for the task can be positively matched to the Dockerfile the command was executed for.

**Comment by bwateratmsft** [comment]
I'm going to leave this issue open as there are a couple of things I want to address.

1. Make the above required parameters actually show as required in the task definition. In fact, I'm not sure any of them actually need to be required per-se, but rather the logic for interpreting defaults should be adjusted.
1. Currently `dockerBuild.dockerfile` must be an absolute path (or have a e.g. `${workspaceFolder}` variable) to match to the Dockerfile when running the "Build Image" command. Ideally if the path is relative, it should be measured relative to the build context.

**Comment by bwateratmsft** [comment]
This change in #3402 is now released in Docker extension version 1.20.0.
